src/algorand_blocks/block_header.rs

```diff
@@ -26,6 +27,20 @@ fn is_zero_hash(hash: &Option<AlgorandHash>) -> bool {
     }
 }
 
+fn is_zero_hash_or_none(hash: &Option<AlgorandHash>) -> bool {
+    match hash {
+        Some(hash) => hash.is_zero(),
+        _ => true,
+    }
+}
+
+fn is_empty_or_none<K, V>(hash_map: &Option<HashMap<K, V>>) -> bool {
```

Comment:
Move to some new `./src/utils.rs` mod (visible only in the crate, not the external API.)

Reply:
You did this whilst I was reviewing. Sorted.